src/main/java/com/github/technus/tectech/thing/metaTileEntity/hatch/GT_MetaTileEntity_Hatch_InputData_Wireless.java

```diff
@@ -0,0 +1,212 @@
+        if (aBaseMetaTileEntity.isClientSide()) {
+            return true;
+        }
+        try {
+            EntityPlayerMP player = (EntityPlayerMP) aPlayer;
+            clientLocale = (String) FieldUtils.readField(player, "translator", true);
+        } catch (Exception e) {
+            clientLocale = "en_US";
+        }
+        return true;
+    }
+
+    @Override
+    public boolean allowPullStack(IGregTechTileEntity aBaseMetaTileEntity, int aIndex, ForgeDirection side,
+            ItemStack aStack) {
+        return false;
+    }
+
+    @Override
+    public boolean allowPutStack(IGregTechTileEntity aBaseMetaTileEntity, int aIndex, ForgeDirection side,
+            ItemStack aStack) {
+        return false;
+    }
+
+    @Override
+    public int getInventoryStackLimit() {
+        return 1;
+    }
+
+    @Override
+    public boolean isOutputFacing(ForgeDirection side) {
+        return false;
+    }
+
+    @Override
+    public boolean isInputFacing(ForgeDirection side) {
+        return side == getBaseMetaTileEntity().getFrontFacing();
+    }
+
+    @Override
+    public void onRemoval() {
+        stacks = null;
+    }
+
+    @Override
+    public int getSizeInventory() {
+        return inventoryDataPacket != null ? inventoryDataPacket.getContent().length : 0;
+    }
+
+    @Override
+    public ItemStack getStackInSlot(int aIndex) {
+        return inventoryDataPacket != null && aIndex < inventoryDataPacket.getContent().length
+                ? inventoryDataPacket.getContent()[aIndex]
+                : null;
+    }
+
+    @Override
+    public boolean shouldDropItemAt(int index) {
+        return false;
+    }
+
+    @Override
+    public void onPreTick(IGregTechTileEntity aBaseMetaTileEntity, long aTick) {
+        super.onPreTick(aBaseMetaTileEntity, aTick);
+        if (MOVE_AT == aTick % 20) {
+            // Retrieve data from the wireless network
+            String networkId = getNetworkIdFromMetaTileEntity();
+            if (networkId != null) {
+                inventoryDataPacket = WirelessDataNetwork.retrieveData(networkId);
+            }
+
+            // Update the active state based on the retrieved data
+            if (inventoryDataPacket == null || inventoryDataPacket.getContent().length == 0) {
+                getBaseMetaTileEntity().setActive(false);
+            } else {
+                getBaseMetaTileEntity().setActive(true);
+            }
+        }
+    }
+
+    public void onFirstTick(IGregTechTileEntity aBaseMetaTileEntity) {
+        if (aBaseMetaTileEntity.isServerSide()) {
+            // On first tick find the player name and attempt to add them to the map.
+
+            // UUID and username of the owner.
+            owner_uuid = aBaseMetaTileEntity.getOwnerUuid().toString();
+            owner_name = aBaseMetaTileEntity.getOwnerName();
+        }
+    }
+
+    private String getNetworkIdFromMetaTileEntity() {
+        return owner_uuid + owner_name; // Placeholder
```

Comment:
Maybe let `WirelessDataNetwork` generate an ID for you (e.g. `WirelessDataNetwork.getNextId()`)